server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1264,7 +1265,10 @@ public Pair<List<? extends Cluster>, Integer> searchForClusters(final ListCluste
         }
 
         if (hypervisorType != null) {
-            sc.setParameters("hypervisorType", hypervisorType);
+            String hypervisorStr = (String) hypervisorType;
+            String hypervisorSearch = HypervisorGuru.HypervisorCustomDisplayName.value()
+                    .equals(hypervisorStr) ? "Custom" : hypervisorStr;
```

Comment:
this logic is too scattered all over the system. should be in `Hypervisor.HypervisorType`